enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,298 @@
+addresses if necessary to find one that it can route to. The addresses will be
+determined by an equivalent of running `hostname -I` on the assisted installer
+live ISO.
+
+### Risks and Mitigations
+
+Communication between the Agent and the assisted installer must be protected
+with TLS. The Agent must be able to trust the assisted installer service so
+that it does not receive rogue provisioning instructions. This is mitigated
+as described in the "TLS Trust" section above.
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Since the on-premise assisted installer is intended for short-term use just to get
+a cluster installed, there is nothing stateful to upgrade or downgrade.
+
+The user will retrieve a newer version, or potentially an older version if
+older versions are offered, by downloading it from cloud.redhat.com.
+
+### Version Skew Strategy
+
+Not applicable. The on-premise modality of assisted installer does not have its
+own versioning.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Running the assisted installer on-premise is not as simple of an experience as
+utilizing the service from cloud.redhat.com. It requires more work from the
+user prior to beginning a cluster installation.
+
+## Alternatives
+
+### Containers Instead of ISO
+
+Instead of distributing the assisted installer as a live ISO, it could be distributed
+as a podman-compatible Pod definition that runs containers directly on the
+user's machine. The user would copy a `podman` command from cloud.redhat.com and run it
+locally.
+
+Advantages include:
+* no need to support a number of potential virtualization platforms
+* exposing the assisted installer's API service from a container may be easier in many cases than exposing it from a VM
+* in the future the live ISO can run on one of the hosts and serve the bootstrap role. This requires an OS on that host, which the live ISO provides.
```

Comment:
shouldn't this be in the downsides section?

Reply:
Maybe I should rephrase this. What I'm trying to say is that in the future, when we want to run AI on one of the hosts that will become a master, we will at that point need to bundle it with a temporary OS. Shipping that with a live ISO makes a lot of sense. So by establishing that pattern now, up-front, we've already done that work and have exactly one way to ship and run AI on-prem. Whether you're running it on bare metal or as a VM on an existing host, you boot an RHCOS live ISO. Thus it's an advantage.

Does that clarify it at all?

Reply:
Nevermind, you're right, I didn't notice this was in the "Alternatives" section. Will move it.